src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -78,10 +72,19 @@ void CancelWork()
                                     return;
                                 }
 
-                                if (activeStatementsResult.ErrorCode != 0)
+                                switch (activeStatementsResult.ErrorCode)
                                 {
-                                    unexpectedError = activeStatementsResult.ErrorCode;
-                                    return;
+                                    case 0:
+                                        break;
+
+                                    case VSConstants.S_FALSE:
+                                    case VSConstants.E_FAIL:
+                                    case (int)DkmExceptionCode.E_INSTRUCTION_NO_SOURCE:
+                                        return;
```

Comment:
This is incorrect. We can't just skip everything and not cancel.

Reply:
Instead we should to set the result like so:
```
builders[runtimeIndex] = ImmutableArray<ActiveStatementDebugInfo>.Empty;
completion.TrySetResult(builders.ToFlattenedImmutableArrayAndFree());
return;
```

I'd also do this on any non-zero error code, instead of cancelling.

---
In reply to: [197935072](https://github.com/dotnet/roslyn/pull/28070#discussion_r197935072) [](ancestors = 197935072)